### PR TITLE
Ads linked to watch add button

### PR DESCRIPTION
**Pull request by timdng10**

### CODE REVIEW

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs [hunk 1]
```diff
@@ -6,15 +6,20 @@
...
 using k = PPGlobal;
+using System.Collections;
 
 public class MainMenu : PPUIElement
 {
     const int STANDARD_DROPDOWN = k.STANDARD_DROPDOWN;
     const int ALT_SINGLE_DROPDOWN = k.ALT_SINGLE_DROPDOWN;
 
-	[Header("External References")]
+    const string ANDROID_ID = "1347448";
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: Is there a reason these hardcoded and not randomly generated GUIDs or some other instance: https://msdn.microsoft.com/en-us/library/system.guid.newguid(v=vs.110).aspx?

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs [hunk 2]
```diff
@@ -6,15 +6,20 @@
 
 using UnityEngine;
 using UnityEngine.UI;
+using UnityEngine.Advertisements;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: I think ideally we should have a separate class to handle advertisements, we shouldn't have the logic stored inside of `MainMenu`

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs [hunk 3]
```diff
@@ -179,11 +184,47 @@ public void OnGiftsClick()
 
     public void OnWatchAdClick()
     {
-        // Disabled
-        EventController.Event(k.GetPlayEvent(k.EMPTY));
+
+        #if UNITY_IOS
+                Advertisement.Initialize(IOS_ID, true);
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: From the documentation, it appears the convention is to let ads auto-initialize if not in Editor: https://docs.unity3d.com/ScriptReference/Advertisements.Advertisement.Initialize.html. Is there a reason we have these init calls here?

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs [hunk 4]
```diff
@@ -179,11 +184,47 @@ public void OnGiftsClick()
...
+        #if UNITY_ANDROID
+                Advertisement.Initialize(ANDROID_ID, true);
+        #endif
+
+        StartCoroutine(ShowAdWhenReady());
+    }
+
+    IEnumerator ShowAdWhenReady()
+    {
+        while (!Advertisement.IsReady("rewardedVideo"))
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: This string should be a `const` value

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs [hunk 5]
```diff
@@ -179,11 +184,47 @@ public void OnGiftsClick()
...
+            yield return null;
+
+        Advertisement.Show("rewardedVideo", new ShowOptions()
+        {
+            resultCallback = result =>
+            {
+                switch (result)
+                {
+                    case ShowResult.Finished:
+                        Debug.Log("Advertisement Finish");
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: I don't think we need debugging feedback unless the advertisement fails

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs [hunk 6]
```diff
@@ -179,11 +184,47 @@ public void OnGiftsClick()
...
+                        break;
+                    case ShowResult.Failed:
+                        Debug.Log("Advertisement Failed");
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: Should likely be `Debug.LogError`

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs (hunk 1 above)
Comment (timdng10) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: These numbers are already defined in the project

Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: Duly noted. See it now in the Unity Ads webpage.

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs (hunk 3 above)
Comment (timdng10) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: For some reason it didn't auto initialize so i initialized it manually

Comment (imann24) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: Can you see whether adding an #if UNITY_EDITOR check around these is workable?

**Comment by gracebarsny** [comment]
@timdng10 Please remember to add commit notes to each PR. There should be a description!

File: Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs (hunk 3 above)
Comment (timdng10) [inline Assets/Scripts/UI/PickupPup/MainMenu/MainMenu.cs]: I decided to go with a regular if statment if(Advertisment.IsInitialized) instead so it doesnt matter if unity auto initializes or not.

File: Assets/Scripts/AdsAndIAP/AdsManager.cs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            return Instance as AdsManager;
+        }
+    }
+
+    #endregion
+
+    const string ANDROID_ID = "1347448";
+    const string IOS_ID = "1347449";
+
+    const string adType = "rewardedVideo";
```
Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: `const` vars should be at the top of the class (above even static accessors)

Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: `const` vars should using screaming caps (`AD_TYPE`)

File: Assets/Scripts/AdsAndIAP/AdsManager.cs [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+            yield return null;
+
+        Advertisement.Show(adType, new ShowOptions()
+        {
+            resultCallback = result =>
+            {
+                switch (result)
+                {
+                    case ShowResult.Finished:
+                        Debug.Log("Advertisement Finish, reward player");
```
Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: Would you mind removing this `Debug` statement?

File: Assets/Scripts/AdsAndIAP/AdsManager.cs [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                        break;
+                    case ShowResult.Failed:
+                        Debug.LogError("Advertisement Failed");
+                        break;
+                    case ShowResult.Skipped:
+                        Debug.Log("Advertisement Skipped, Do not reward player");
```
Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: Would you mind removing this `Debug` statement?

File: Assets/Scripts/AdsAndIAP/AdsManager.cs [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+                        break;
+                    default:
```
Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: Can you remove this empty `default` case?

File: Assets/Scripts/AdsAndIAP/AdsManager.cs [hunk 11]
```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections;
```
Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: Can you add a header here including `Author` and `Description` (see other scripts for the comment format)

File: Assets/Scripts/AdsAndIAP/AdsManager.cs [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Advertisements;
+
+public class AdsManager : SingletonController<AdsManager> {
```
Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: Opening '{' should be on a line of its own

File: Assets/Scripts/AdsAndIAP/AdsManager.cs [hunk 13]
```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections;
+using System.Collections.Generic;
```
Comment (imann24) [inline Assets/Scripts/AdsAndIAP/AdsManager.cs]: Can you remove any namespaces from the class header which aren't being used by the class?

**Comment by imann24** [comment]
@timdng10 I had a few more small stylistic revisions which I thought it quicker to push directly. Looks good!

**Pull request merged**